src/Components/Blazor/Blazor/src/Hosting/WebAssemblyHost.cs

```diff
@@ -5,94 +5,166 @@
             GC.KeepAlive(typeof(EntrypointInvoker));
             GC.KeepAlive(typeof(JSInteropMethods));
             GC.KeepAlive(typeof(WebAssemblyEventDispatcher));
 
-            Services = services ?? throw new ArgumentNullException(nameof(services));
-            _runtime = runtime ?? throw new ArgumentNullException(nameof(runtime));
+            _services = services;
+            _scope = scope;
+            _configuration = configuration;
+            _rootComponents = rootComponents;
         }
 
-        public IServiceProvider Services { get; }
+        /// <summary>
+        /// Gets the application configuration.
+        /// </summary>
+        public IConfiguration Configuration => _configuration;
+
+        /// <summary>
+        /// Gets the service provider associated with the application.
+        /// </summary>
+        public IServiceProvider Services => _scope.ServiceProvider;
 
-        public Task StartAsync(CancellationToken cancellationToken = default)
+        /// <summary>
+        /// Disposes the host.
+        /// </summary>
+        public void Dispose()
         {
-            return StartAsyncAwaited();
+            if (_disposed)
+            {
+                return;
+            }
+
+            _disposed = true;
+
+            _renderer?.Dispose();
+
+            _scope?.Dispose();
+
+            (_services as IDisposable)?.Dispose();
         }
 
-        private async Task StartAsyncAwaited()
+        /// <summary>
+        /// Disposes the host asynchronously.
+        /// </summary>
+        /// <returns>A <see cref="ValueTask"/> which respresents the completion of disposal.</returns>
+        public async ValueTask DisposeAsync()
         {
-            var scopeFactory = Services.GetRequiredService<IServiceScopeFactory>();
-            _scope = scopeFactory.CreateScope();
-
-            try
+            if (_disposed)
             {
-                var startup = _scope.ServiceProvider.GetService<IBlazorStartup>();
-                if (startup == null)
-                {
-                    var message =
-                        $"Could not find a registered Blazor Startup class. " +
-                        $"Using {nameof(IWebAssemblyHost)} requires a call to {nameof(IWebAssemblyHostBuilder)}.UseBlazorStartup.";
-                    throw new InvalidOperationException(message);
-                }
+                return;
+            }
 
-                // Note that we differ from the WebHost startup path here by using a 'scope' for the app builder
-                // as well as the Configure method.
-                var builder = new WebAssemblyBlazorApplicationBuilder(_scope.ServiceProvider);
-                startup.Configure(builder, _scope.ServiceProvider);
+            _disposed = true;
 
-                _renderer = await builder.CreateRendererAsync();
+            _renderer?.Dispose();
+
+            if (_scope is IAsyncDisposable asyncDisposableScope)
```

Comment:
Yeah you're right this is a bug. I'm going to just remove `IDisposable` from this class. 

BTW the decision we made for this in general (see DI) is that we block if someone calls `Dispose()`.

Since this is new code we don't have to implement `IDisposable` at all.